event_response_tool/layer_loss_duplicator/duplicate_layer_loss.py

```diff
@@ -0,0 +1,348 @@
+                f"Exception occured while scaling Loss Set: {loss_set.id}"
+            )
+        else:
+            return weighted_elt_df
+
+    def transform_loss_set(self, loss_set):
+        alert.debug(f"Transforming loss_set {loss_set.id}")
+        old_elt_data = loss_set.download_data()
+        elt_df = read_byte_stream_into_csv(old_elt_data)
+        # Scale and transform loss set data and save into string buffer
+        scaled_df = self.scale_elt(elt_df, loss_set)
+        if scaled_df is not None:
+            new_description = (f"ER__{loss_set.description}").replace(" ", "")
+            new_loss_set = self.upload_elt(
+                new_description,
+                scaled_df.to_csv(index=False),
+                loss_set.currency,
+                loss_set.event_catalogs,
+            )
+            return new_loss_set
+
+    def upload_elt(self, description, elt, currency, catalogs):
+        try:
+            loss_set = analyzere.LossSet(
+                type="ELTLossSet",
+                description=description,
+                event_catalogs=catalogs,
+                currency=currency,
+                loss_type=self.config.get("defaults", "loss_perspective"),
+            ).save()
+            loss_set.upload_data(elt)
+            check_resource_upload_status(loss_set)
+        except Exception as e:
+            alert.exception(
+                f"Exception occured while uploading ELT {loss_set.id}: {e}"
+            )
+        if loss_set.status == "processing_succeeded":
+            alert.debug(f"Uploaded loss_set {loss_set.id}")
+        else:
+            alert.error(
+                f"LossSet {loss_set.id} was uploaded, but failed while processing. {loss_set.status_message}"
+            )
+
+        return loss_set
+
+    def modify_layer(self, layer):
+        # Recursively process the layer structure and transform loss sets
+        if hasattr(layer, "ref_id"):
+            layer = analyzere.LayerView.retrieve(layer.ref_id).layer
+        if hasattr(layer, "analysis_profile"):
+            layer = layer.layer
+        if type(layer) in [analyzere.LayerView, Reference]:
+            layer = layer.layer
+        if layer.type == "BackAllocatedLayer":
+            logger.exception("Back Allocated layers are not supported")
+            raise
+
+        if layer.type == "NestedLayer":
+            layer.sink = self.modify_layer(layer.sink)
+            layer.sources = [self.modify_layer(s) for s in layer.sources]
+
+        if hasattr(layer, "loss_sets"):
+            try:
+                # no losses to process
+                if len(layer.loss_sets) == 0 and layer.type != "FilterLayer":
+                    pass
+                # Need to replace Filter Layers with unlimited Generic layer
+                elif layer.type == "FilterLayer":
+                    # Transform any loss sets that the Filter Layer may have
+                    loss_sets = []
+
+                    for loss_set in layer.loss_sets:
```

Comment:
the non-filter layer case has extra loss set handling steps, could this be restructured to do the same loss set transformation for both?